tests/chainer_tests/utils_tests/test_precision.py

```diff
@@ -0,0 +1,58 @@
+import unittest
+
+import numpy
+
+from chainer import function_node
+from chainer import testing
+from chainer import utils
+
+
+class F(function_node.FunctionNode):
+
+    @utils.mixed_precision
+    def forward(self, x):
+        self.x = x
+        return x
+
+
+class TestMixedPrecision(unittest.TestCase):
+
+    def test_fp16(self):
+        x = (numpy.zeros((1, 2, 3), dtype=numpy.float16),) * 2
+        f = F()
+        y = f.apply(x)
+        assert f.x[0].dtype == numpy.float32
+        assert f.x[1].dtype == numpy.float32
+        assert y[0].dtype == numpy.float16
+        assert y[1].dtype == numpy.float16
+
+    def test_fp32(self):
+        x = (numpy.zeros((1, 2, 3), dtype=numpy.float32),) * 2
+        f = F()
+        y = f.apply(x)
+        assert f.x[0] is x[0]
+        assert f.x[1] is x[1]
```

Comment:
This test is weak since `x[0]` is `x[1]`.

Reply:
oh...

Reply:
Fixed!